### PR TITLE
AP_RangeFinder: LeddarOne remove clear buffer in send_request

### PR DESCRIPTION
**Pull request by ShingoMatsuura**
remove clear buffer in send_request function. 

checked by logging whether available bytes exist. There were no available bytes.

### CODE REVIEW

**Comment by magicrub** [comment]
all it takes is to be out of sync with one byte to mess up this system - it's fragile. How about this:

1. in case LEDDARONE_MODBUS_STATE_INIT you perform the uart read flush
2. in case LEDDARONE_MODBUS_STATE_AVAILABLE when calling parse_response() you handle result LEDDARONE_READ_BUFFER_SIZE (and other errors) by setting state to LEDDARONE_MODBUS_STATE_INIT

That will allow the system to reset and recover from any scenario.

**Comment by ShingoMatsuura** [comment]
Thank you for reviewing @magicrub - san. Im going to reconsider these case.

**Comment by magicrub** [comment]
@ShingoMatsuura while you're at it, please catch all the other errors and set to INIT. Please consider the condition where there's a loose wire and the stream is often corrupted. The parsing must be able to recover from any random intermittent byte stream.

**Comment by ShingoMatsuura** [comment]
Add clear buffer at INIT state.

changed send buffer variable to member variable. the send buffer is always same. Changing to member variable is to reduce initialize and CRC computation.

File: libraries/AP_RangeFinder/AP_RangeFinder_LeddarOne.cpp [hunk 1]
```diff
@@ -50,13 +62,24 @@ bool AP_RangeFinder_LeddarOne::get_reading(uint16_t &reading_cm)
...
+    uint8_t index = 0;
+    uint32_t nbytes;
 
     if (uart == nullptr) {
         return false;
     }
 
     switch (modbus_status) {
     case LEDDARONE_MODBUS_STATE_INIT:
+        // clear read buffer
```
Comment (magicrub) [inline libraries/AP_RangeFinder/AP_RangeFinder_LeddarOne.cpp]: if you wrap the inside of this case with { } then you can declare uint8_t and uint32_t inside the case.

File: libraries/AP_RangeFinder/AP_RangeFinder_LeddarOne.h [hunk 2]
```diff
@@ -74,4 +74,6 @@ class AP_RangeFinder_LeddarOne : public AP_RangeFinder_Backend
     LeddarOne_ModbusStatus modbus_status = LEDDARONE_MODBUS_STATE_INIT;
     uint8_t read_buffer[LEDDARONE_READ_BUFFER_SIZE];
     uint32_t read_len;
+
+    uint8_t send_request_buffer[8] = {0};
```
Comment (magicrub) [inline libraries/AP_RangeFinder/AP_RangeFinder_LeddarOne.h]: I like this moving here to save the work but would be even more helpful if it was const. Is that possible with the CRC calc in the constructor? Maybe if the crc was static?

**Comment by magicrub** [comment]
Sorry to keep nagging you on this, I just keep finding ways to make it better but since I don't have the hardware its safer if you make the changes and test since I can't. Otherwise I'd make them myself :)

At line:
>else if (leddarone_status != LEDDARONE_STATE_READING_BUFFER)

can you add an || timeout check like you have here:
>if (AP_HAL::millis() - last_sending_request_ms > 200)

Also, that else if timeout that's already there can be one the same line, no need for the nested { }.

**Comment by ShingoMatsuura** [comment]
Thank you for reviewing ＠magicrub - san. It is not nagging:) I appreciate your reviewing. I am going to try fix regarding your advice.

**Comment by muramura** [comment]
The send_request_buffer variable is used only for the send_request method.
In the constructor, CRC is not set in send_request_buffer.
Therefore,
In send_request method it is better to set a fixed value for static send_request_buffer variable.

**Comment by magicrub** [comment]
... hence my request to make send_request_buffer a const

**Comment by ShingoMatsuura** [comment]
Thank you for comment @muranura - san, @magicrub - san. lm going to change send_request_buffer to const.

**Comment by magicrub** [comment]
Lets move the const buffer definition to the header and then this is ready to be merged. Thanks for being so quick on the changes!

**Comment by ShingoMatsuura** [comment]
Thank you for reviewing @magicrub - san. I fixed them.

**Pull request merged**